lib/chef/provider/package/rubygems.rb

```diff
@@ -533,18 +533,21 @@ def gem_binary_path
         end
 
         def install_via_gem_command(name, version)
-          if new_resource.source =~ /\.gem$/i
+          src = []
+          if new_resource.source.is_a?(String) && new_resource.source =~ /\.gem$/i
             name = new_resource.source
-          elsif new_resource.clear_sources
-            src = " --clear-sources"
-            src << (new_resource.source && " --source=#{new_resource.source}" || "")
           else
-            src = new_resource.source && " --source=#{new_resource.source} --source=#{Chef::Config[:rubygems_url]}"
+            src << "--clear-sources" if new_resource.clear_sources
+            srcarry = [ new_resource.source || Chef::Config[:rubygems_url] ].flatten.compact
+            srcarry.each do |s|
+              src << "--source=#{s}"
```

Comment:
Should technically use `Shellwords.escape` here on Unix but I don't want to think of how that would break on Windows or how terrible this would be in general with metachars in a source URL. Can heap this on the "should be using array form of shellout" debt pile.